### PR TITLE
Update the contributing docs

### PR DESCRIPTION
**Pull request by aeisenberg**
Just adds some more details.

## Checklist

- [n/a] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [n/a] Issues have been created for any UI or other user-facing changes made by this pull request.
- [n/a] `@github/docs-content-dsp` has been cc'd in all issues for UI or other user-facing changes made by this pull request.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -93,13 +93,25 @@ Alternatively, you can run the tests inside of vscode. There are several vscode
...
+    * Create a new commit with a message the same as the branch name.
+    * Create a PR for this branch.
+    * Wait for the PR to be merged into `main`
+1. Trigger a release build on Actions by adding a new tag on branch `main` named after the release, as above. Note that when you push to upstream, you will need to fully qualify the ref. A command like this will work:
+
+    ```bash
+    git push upstream refs/tags/v1.3.6
+    ```
+
+    * **IMPORTANT** Make sure you are on the master branch and you are fully updated when you add the tag.
```
Comment (adityasharad) [inline CONTRIBUTING.md:109]:
```suggestion
    * **IMPORTANT** Make sure you are on the `main` branch and your local checkout is fully updated when you add the tag.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -93,13 +93,25 @@ Alternatively, you can run the tests inside of vscode. There are several vscode
...
-   be correct.
-1. Trigger a release build on Actions by adding a new tag on branch `main` of the format `vxx.xx.xx`
+1. Double-check that the extension `package.json` has the version you intend to release. If you are doing a patch release (as opposed to minor or major version) this should already be correct.
+1. Create a PR for this release:
+    * This PR will contain any missing bits from steps 1 and 2. Most of the time, this will just be updating `CHANGELOG.md` with today's date.
+    * Create a new branch for the release named after the new version. For example: `v1.3.6`
+    * Create a new commit with a message the same as the branch name.
+    * Create a PR for this branch.
+    * Wait for the PR to be merged into `main`
+1. Trigger a release build on Actions by adding a new tag on branch `main` named after the release, as above. Note that when you push to upstream, you will need to fully qualify the ref. A command like this will work:
```
Comment (adityasharad) [inline CONTRIBUTING.md:103]: Future: would it be useful to have a workflow dispatch event for this too? Then you can release from the UI.

**Review by adityasharad — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (aeisenberg) [reply CONTRIBUTING.md:103]: Possibly, though by doing it locally, you are (I am) more likely to go over everything in detail.

File: CONTRIBUTING.md (hunk 1 above)
Comment (aeisenberg) [reply CONTRIBUTING.md:109]: Yes.

**Pull request merged**